### PR TITLE
feat: Improved Data Validation for Fiber Content Calculation

### PR DESCRIPTION
**Pull request by Payne680**
… value

### What

 Comprehensive Validation to Ensure Soluble Fiber Plus Insoluble Fiber Does Not Exceed Total Fiber Content

Description:
This pull request aims to enhance the existing data validation process by implementing a more detailed and comprehensive check for the calculation of fiber content. Specifically, this enhancement focuses on validating the relationship between soluble fiber, insoluble fiber, and the total fiber content of a product.
### Screenshot

Screenshot 2024-03-14 at 16 37 01

### Related issue(s) and discussion

- Added a validation condition to check if the sum of soluble fiber and insoluble fiber exceeds the fiber value.
- If the condition is met, the error tag "en:nutrition-soluble-fiber-plus-insoluble-fiber-greater-than-fiber" is added to the data_quality_errors_tags array.

- Fixes #[ISSUE NUMBER]
#9729

### CODE REVIEW

File: lib/ProductOpener/DataQualityFood.pm [hunk 1]
```diff
@@ -1302,6 +1302,27 @@ sub check_nutrition_data ($product_ref) {
...
 		}
 
+
+# sum of nutriments that compose fiber can not be greater than fiber value
+if (
+    (defined $product_ref->{nutriments}{fiber_100g})
+    and (
+        (
+            (
+                (defined $product_ref->{nutriments}{soluble_fiber_100g}) ? $product_ref->{nutriments}{soluble_fiber_100g}
```
Comment (stephanegigandet) [inline lib/ProductOpener/DataQualityFood.pm:1312]:
```suggestion
                (defined $product_ref->{nutriments}{"soluble-fiber_100g"}) ? $product_ref->{nutriments}{soluble_fiber_100g}
```

It's difficult to spot, but if you look at the JSON you got for your product, the field name is soluble-fiber_100g (with a dash between soluble and fiber) and not soluble_fiber_100g

Same change is needed for insoluble fiber

**Comment by stephanegigandet** [comment]
Thanks @Payne680 ! To test your code, you should add a test case in https://github.com/openfoodfacts/openfoodfacts-server/blob/main/tests/unit/dataqualityfood.t 

You can then run the tests using "make test-unit test=dataqualityfood.t"

File: lib/ProductOpener/DataQualityFood.pm (hunk 1 above)
Comment (Payne680) [reply lib/ProductOpener/DataQualityFood.pm:1312]: @stephanegigandet Thanks for the review, I have corrected the error. And am getting the correct error message when soluble fiber is plus is insoluble fiber is greater than the fiber value.
Screenshot 2024-03-14 at 18 32 08

**Comment by Payne680** [comment]
> Thanks @Payne680 ! To test your code, you should add a test case in https://github.com/openfoodfacts/openfoodfacts-server/blob/main/tests/unit/dataqualityfood.t
> 
> You can then run the tests using "make test-unit test=dataqualityfood.t"

Yes @stephanegigandet, am left with writing the test cases before moving the draft PR to a normal PR

File: html/off_web_html/html [hunk 2]
```diff
@@ -0,0 +1 @@
+/path/to/openfoodfacts-web/html
```
Comment (stephanegigandet) [inline html/off_web_html/html:1]: I think this was added to the PR by mistake as it seems unrelated, can you remove it?

**Comment by stephanegigandet** [comment]
@Payne680 It looks good to me, thank you! Can you change the title of the PR "feat: Work still in progress" to something more descriptive of what the change is and then remove the Draft status?

File: html/off_web_html/html (hunk 2 above)
Comment (Payne680) [reply html/off_web_html/html:1]: Yes @stephanegigandet, I will remove it right away

**Review by stephanegigandet — APPROVED** [review]
Looks good, thank you!

**Comment by stephanegigandet** [comment]
@Payne680 to fix the formatting style, you can just run "make lint", and then commit any changes
